update/Pos/Update/Poll/Types.hs

```diff
@@ -211,6 +213,9 @@ data PollModifier = PollModifier
     , pmActiveProps       :: !(MapModifier UpId ProposalState)
     , pmDelActivePropsIdx :: !(HashMap ApplicationName (HashSet UpId))
     , pmSlottingData      :: !(Maybe SlottingData)
+    , pmEpochProposers    :: !(Maybe (HashSet StakeholderId))
+    -- ^ Cache of DB state.
```

Comment:
I read the code and I see why you call it cache. However, I don't like this approach because:
· I don't like that you are mixing actual GState (well, cache of this state) and modifier in one type. I think it makes code less understandable. It's clearer when modifier contains only modifier.
· As a consequence, if we don't modify epoch proposers, but only read them, we will still write proposers to DB. AFAIU, it's impossible now, so it's not important.
· Later we will load all US GState into RAM and will store it in DB only for persistent. US GState should be small (need to calculate) unlike Txp state, so it's fine to fully load it into memory. So all state will be stored in another data type.